socket.c

```diff
@@ -36,24 +36,7 @@
 static const int YES = 1;
 static const int TTL = 255;
 
-static int socket_bind_to_device(int sock, const char *iface) {
```

Comment:
I think we should still try keep this to ensure we only receive packets from this interface, even if we could drop any non-matching packets later based on `iface_ifindex`. But should change it to not fail if setsockopt fails and rather just emit a warning.

Reply:
Sounds good.  I've pushed an update, let me know what you think.  Thanks!